h2o-py/tests/testdir_algos/common/pyunit_train_preparation_perf.py

```diff
@@ -0,0 +1,51 @@
+import sys
+import time
+sys.path.insert(1, "../../../")
+
+import h2o
+from h2o.estimators.estimator_base import H2OEstimator
+from tests import pyunit_utils as pu
+
+
+class DummyEstimator(H2OEstimator):
+    
+    def __init__(self):
+        super(DummyEstimator, self).__init__()
+        self._parms = {}
+
+    def _train(self, parms, verbose=False):
+        pass
+
+
+def test_basic_estimator_preparation_is_fast_enough():
+    dummy = DummyEstimator()
+    # for some mysterious reason, the parser fails if uploading more than 331186 columns: magic number?
+    # using this upper limit for now, but this is fishy:
+    #
+    # ~/repos/h2o/h2o-3/h2o-py/h2o/h2o.py in parse_setup(raw_frames, destination_frame, header, separator, column_names, column_types, na_strings, skipped_columns, custom_non_data_line_markers, partition_by, quotechar, escapechar)
+    #     874             if len(column_names) != len(j["column_types"]): raise ValueError(
+    #     875                 "length of col_names should be equal to the number of columns: %d vs %d"
+    # --> 876                 % (len(column_names), len(j["column_types"])))
+    #     877         j["column_names"] = column_names
+    #     878         counter = 0
+    # 
+    # ValueError: length of col_names should be equal to the number of columns: 1000000 vs 331186
```

Comment:
@michalkurka is it a known limitation? is it worth a ticket?

Reply:
I will try to reproduce this

Reply:
There is a magic number, the default chunk size for Uploaded data is 4MB, your file is about 22MB, and it is split into 6 chunks. It looks like the header is longer than one chunk and the data is incorrectly parsed.

Something like this should only happen for datasets like yours, super-short and ultra-wide.

It is a bug that should be documented.

Reply:
documented in https://h2oai.atlassian.net/browse/PUBDEV-8898